wire-ios-utilities/Source/NSManagedObjectContext+PerformGrouped.swift

```diff
@@ -65,3 +65,21 @@ public extension NSManagedObjectContext {
         }
     }
 }
+/**
+ Wrapper around Task to make sure tests are waiting for the task to be finished using dispatchGroups attached to NSManagedObjectContext. See `WaitForAllGroupsToBeEmpty()`
+ */
+public struct ZMTask {
+    let context: NSManagedObjectContext
+
+    public init(context: NSManagedObjectContext) {
+        self.context = context
+    }
+
+    public func callAsFunction(_ block: @escaping () async -> Void) {
+        let groups = context.enterAllGroupsExceptSecondary()
+        Task {
+            await block()
+            context.leaveAllGroups(groups)
+        }
+    }
+}
```

Comment:
To be honest I didn't know what `ZMTask` is supposed to do and why it is needed. In Swift we are not supposed to use prefixes like `ZM`.

For me this looks like we don't use patterns around `async await` and `GCD` as intended, but maybe it just some case I've never seen. What is the intention of `enterAllGroupsExceptSecondary`? From the implementation I have a lot of questions.

Do we need a shared `struct` to make it work? It feels like something that we should get rid such a solution mixing GCD and Swift concurrency as soon as possible. Can the few lines not be directly implemented in this way where it is used?

Reply:
The IntegrationTests depend a lot on this the dispatchGroup associated to the contexts in order to wait for things to be finished.

Introducing `Task` in the codebase started to break these tests, since we don't wait for the tasks to be complete.
This helper (maybe the naming `ZMTask` is poorly chosen) is a way to enforce waiting for these tasks.

some places directly use the groups enter and leave, this is an attempt to make it clear and **easier** to remove once we found a better solution (or replaced most of the integration tests by unit tests)

Reply:
After chatting I understand better why this is needed, for testing. Though the code is also executed in production without any effects.
My suggestion would be to find a more descriptive name, maybe `ManagedObjectContextTask` or `WaitingContextTask`?